osu.Game.Rulesets.Osu/Difficulty/OsuPerformanceCalculator.cs

```diff
@@ -400,45 +399,44 @@ private double calculateEstimatedSliderBreaks(double topWeightedSliderFactor, Os
-            const double z = 2.32634787404; // 99% critical value for the normal distribution (one-tailed).
-
-            // Proportion of greats hit on circles, ignoring misses and 50s.
+            // The sample proportion of successful hits.
+            double n = Math.Max(1, relevantCountGreat + relevantCountOk);
             double p = relevantCountGreat / n;
 
-            // We can be 99% confident that p is at least this value.
-            double pLowerBound = (n * p + z * z / 2) / (n + z * z) - z / (n + z * z) * Math.Sqrt(n * p * (1 - p) + z * z / 4);
+            // 99% critical value for the normal distribution (one-tailed).
+            const double z = 2.32634787404;
 
-            // Compute the deviation assuming greats and oks are normally distributed, and mehs are uniformly distributed.
-            // Begin with greats and oks first. Ignoring mehs, we can be 99% confident that the deviation is not higher than:
-            double deviation = greatHitWindow / (Math.Sqrt(2) * DifficultyCalculationUtils.ErfInv(pLowerBound));
+            // We can be 99% confident that the population proportion is at least this value.
+            double pLowerBound = Math.Max(p, (n * p + z * z / 2) / (n + z * z) - z / (n + z * z) * Math.Sqrt(n * p * (1 - p) + z * z / 4));
 
-            double randomValue = Math.Sqrt(2 / Math.PI) * okHitWindow * Math.Exp(-0.5 * Math.Pow(okHitWindow / deviation, 2))
-                                 / (deviation * DifficultyCalculationUtils.Erf(okHitWindow / (Math.Sqrt(2) * deviation)));
+            double deviation;
 
-            deviation *= Math.Sqrt(1 - randomValue);
+            if (pLowerBound > 0)
+            {
+                // Compute deviation assuming greats and oks are normally distributed.
+                deviation = greatHitWindow / (Math.Sqrt(2) * DifficultyCalculationUtils.ErfInv(pLowerBound));
 
-            // Value deviation approach as greatCount approaches 0
-            double limitValue = okHitWindow / Math.Sqrt(3);
+                // Subtract the deviation provided by tails that land outside the ok hit window from the deviation computed above.
+                // This is equivalent to calculating the deviation of a normal distribution truncated at +-okHitWindow.
+                double okHitWindowTailAmount = Math.Sqrt(2 / Math.PI) * okHitWindow * Math.Exp(-0.5 * Math.Pow(okHitWindow / deviation, 2))
+                                               / (deviation * DifficultyCalculationUtils.Erf(okHitWindow / (Math.Sqrt(2) * deviation)));
 
-            // If precision is not enough to compute true deviation - use limit value
-            if (Precision.AlmostEquals(pLowerBound, 0.0) || randomValue >= 1 || deviation > limitValue)
-                deviation = limitValue;
+                deviation = okHitWindowTailAmount < 0.99 ? deviation * Math.Sqrt(1 - okHitWindowTailAmount) : okHitWindow / Math.Sqrt(3);
```

Comment:
`okHitWindowTailAmount < 0.99` will definitely change values
also, it can probably add big jump going from something like 350UR right to limit value of 600UR with 1x100
I don't sure how severe it is, but it definitely should be tested

Reply:
also, the line with effect `deviation = Math.Max(deviation, limitValue)` removed and probably should be readded
I don't sure it can ever trigger but better be safe
in the current code this line handles all the cases where `okHitWindowTailAmount` is lower than 1 but very close to it (instead of 0.99 limit)

Reply:
For your first reply, I have tested and it does not cause any jumps going from 99% to the limit value. Both result in the same UR, no matter the hit window size. You can test it [here](https://www.desmos.com/calculator/5enkmnrgyw). ![image](https://github.com/user-attachments/assets/50ab3405-a81a-49b3-822d-77be85cbea09)

For your second reply, no it can never trigger due to the stated reason above. As well, this case (it being very close to 0.99) requires there to be 25 100s for every 1 300. A very rare case.